controller/src/solenoid.cpp

```diff
@@ -16,6 +16,11 @@ limitations under the License.
 #include "solenoid.h"
 
 #include "hal.h"
+#include "parameters.h"
+#include "stdint.h"
+
+// Solenoid output pin 5 TBC.
+static const int O_SOLENOID = 5;
```

Comment:
Also, just curious where this came from? There's documentation somewhere?

Reply:
I genuinely have no idea about this one; probably a question for Edwin.